File: packages/typia/src/transformers/features/protobuf/ProtobufValidateDecodeTransformer.ts
import { ProtobufValidateDecodeProgrammer } from "../../../programmers/protobuf/ProtobufValidateDecodeProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufValidateDecodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "protobuf.validateDecode",
      write: ProtobufValidateDecodeProgrammer.write,
    });
}

File: packages/typia/native/transform/features/protobuf/ProtobufValidateDecodeTransformer.go
package protobuf

import (
  shimast "github.com/microsoft/typescript-go/shim/ast"
  nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
  nativeprotobufprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/protobuf"
  nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type protobufValidateDecodeTransformerNamespace struct{}

var ProtobufValidateDecodeTransformer = protobufValidateDecodeTransformerNamespace{}

func (protobufValidateDecodeTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
  return nativeinternal.GenericTransformer.Scalar(nativeinternal.GenericTransformer_IProps{
    ITransformProps: props,
    Method:          "protobuf.validateDecode",
    Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
      return nativeprotobufprogrammers.ProtobufValidateDecodeProgrammer.Write(nativeprotobufprogrammers.ProtobufValidateDecodeProgrammer_IProps(x))
    },
  })
}
